src/memory/mod.rs

```diff
@@ -42,14 +46,7 @@ impl MemoryManagers {
 
 impl<T: MemoryManagerUpdate> MemoryManager<T> {
     fn ready_for_updates(&mut self, _ctx: &StateContext) -> bool {
-        if let Some(class) = self.manager.singleton {
```

Comment:
Not entirely sure what is changed here and why. We are checking if there is a singleton, and that means we are ready for updates?

Reply:
we no longer need to check class == 0 because of some earlier updates, and clippy recommended this change when i removed the if statement :)

Basically if the singleton isn't None, we should be good to go, or the previous update_manager call would have reset it.